harvard-lts/etd-base-template
# ETD-193: add logging to template, now Dockerized and w/ fixed tracing

**The title of this pull-request should be a brief description of what the pull-request fixes/improves/changes. Ideally 50 characters or less.**
* * *

**JIRA Ticket**: https://jira.huit.harvard.edu/browse/ETD-193


* Other Relevant Links (Mailing list discussion, related pull requests, etc.)

# What does this Pull Request do?
adds logging, fixes tracing, dockerizes template

# How should this be tested?

1) copy example.env to .env and populate values
2) run jaeger locally according to this wiki: https://wiki.harvard.edu/confluence/display/LibraryTechServices/OpenTelemetry
3) start by running : `docker-compose -f docker-compose-local.yml up -d --build --force-recreate`
4) enter container: `docker exec -it etd-base-template bash`
5) run `pytest`. all tests should pass.
6) run test script: `python3 scripts/invoke-task.py`
7) `cd ./logs/etdadm/etd` and look for the latest log. there should be log data.
8) go to local jaeger at http://localhost:16686/   then search for ETD in the service. you should see a trace from that test.
9) stop container: `docker-compose -f docker-compose-local.yml down `

# Test coverage
Yes/No: Are changes in this pull-request covered by:
- unit tests? y
- integration tests? y

# Interested parties
Tag (@ mention) interested parties: @awoods
